src/main/java/org/fogbowcloud/arrebol/ArrebolController.java

```diff
@@ -75,39 +70,65 @@ public ArrebolController(Properties properties) {
     private static final String RAW_TYPE = "raw";
     private static final String DOCKER_TYPE = "docker";
 
-    private WorkerPool createPool(Properties properties, int poolId) {
+    private WorkerPool createPool(Configuration configuration, int poolId) {
 
         //we need to deal with missing/wrong properties
 
         Collection<Worker> workers = new LinkedList<>();
-        String poolType = properties.getProperty("pool.type");
-
-        int poolSize = new Integer(properties.getProperty("pool.size"));
-        for (int i = 0; i < poolSize; i++) {
-            TaskExecutor executor = createTaskExecutor(poolType, properties);
-            Specification resourceSpec = null;
-            Worker worker = new MatchAnyWorker(resourceSpec, "resourceId-"+i, poolId, executor);
-            workers.add(worker);
-        }
+        populatePool(workers, poolId, configuration);
 
         WorkerPool pool = new StaticPool(poolId, workers);
         LOGGER.info("pool={" + pool + "} created with workers={" + workers + "}");
 
         return pool;
     }
 
-    private TaskExecutor createTaskExecutor(String type, Properties properties) {
+    private void populatePool(Collection<Worker> workers, int poolId, Configuration configuration){
+        String poolType = configuration.getPoolType();
+        if(poolType.equals(DOCKER_TYPE)){
```

Comment:
As the Arrebol worker's type may increase, this if will increase too, so I think that could be more elegant to use the strategy pattern.

Reply:
This was solved in 26d2e10